clustered/server/src/main/java/org/ehcache/clustered/server/ServerStoreImpl.java

```diff
@@ -66,11 +67,13 @@ public Chain get(long key) {
 
   @Override
   public void append(long key, ByteBuffer payLoad) {
+    checkPayLoadSize(payLoad);
```

Comment:
I will let someone else (@chrisdennis) confirm but it feels weird that it's not the server that is responsible to verify this. Also, why can't it happen on `replaceAtHead`, and `put`?

Reply:
@henri-tremblay this is just to prevent a very large put (larger than the cache) act like a cache.clear().. Ofcourse, we may still have a very large put that fits the cache that may evict everything else..but that may be a legitimate thing..

Since we prevent such chains being formed, replaceAtHead (which is not a direct client action) and put (passive only) should not have such large single chain that crosses the pool size on the active and passive.

It is debatable whether to put this check on client side or server side. One advantage of server side is we have got all the overhead in the payload and it is easier to check pool size.

Reply:
The advantage of client-side you mean?

Reply:
no currently it is server side.. as ServerStoreImpl is on server side..

Reply:
Oh dear... Totally misread. Sorry. All good then.